inst/shiny/modules/tab_visuals.R

```diff
@@ -0,0 +1,533 @@
+# The Visualisation Navbar tab loads the data from the Data tab, and results from NCA tab
+# The user can then explore the data using various visualisation tools
+
+# VISUALISATION ----
+tab_visuals_ui <- function(id) {
+  ns <- NS(id)
+
+  navset_card_pill(
+    header = "Exploratory Analysis",
+    id = "visuals",
+    nav_panel("Individual Plots",
+      layout_sidebar(
+        sidebar = sidebar(
+          position = "right", open = TRUE,
+          pickerInput(
+            inputId = ns("generalplot_analyte"),
+            label = "Select Analyte:",
+            choices = NULL,
+            selected = NULL,
+            multiple = FALSE,
+            options = list(`actions-box` = TRUE)
+          ),
+          pickerInput(
+            inputId = ns("generalplot_usubjid"),
+            label = "Select Subjects:",
+            choices = NULL,
+            selected = NULL,
+            multiple = TRUE,
+            options = list(`actions-box` = TRUE)
+          ),
+          pickerInput(
+            inputId = ns("generalplot_colorby"),
+            label = "Choose the variables to color by",
+            choices = NULL,
+            selected = NULL,
+            multiple = TRUE,
+            options = list(`actions-box` = TRUE)
+          ),
+          radioButtons(
+            ns("log"),
+            "Select the Plot type:",
+            choices = c("Lin", "Log")
+          ),
+          radioButtons(
+            ns("timescale"),
+            "Choose the Timescale",
+            choices = c("All Time", "By Cycle")
+          ),
+          conditionalPanel(
+            condition = "input.timescale == 'By Cycle'",
+            uiOutput(ns("cycleselect")),
+            ns = NS(id)
+          )
+        ),
+        plotlyOutput(ns("individualplot"))
+      )
+    ),
+    nav_panel("Mean Plots",
+      layout_sidebar(
+        sidebar = sidebar(
+          selectInput(
+            inputId = ns("analytemean"),
```

Comment:
nitpick: Would be nice to clean up IDs to be more readable and break up this kind of pattern:

`analytemean` -> `analyte_mean`
`selectidvar` -> `select_id_var`

and maybe be more verbose with naming some of the ids:

`paramselect` -> `select_display_parameters`

It is kinda hard to read IDs like `selected_varvalstofilter_boxplot`.